### ISSUE TITLE
--skip-bav doesn't skip open redirect scanning

### ISSUE THREAD

**Issue by pdelteil** [hahwul/dalfox#215]
## Describe the bug

As I understand using the flag --skip-bav should disable the scanning of open redirect. While testing I discovered that this scanning is still being done (I think)

```
> dalfox url 'https://ditu.google.com' --skip-bav
[*] 🦊 Start scan [SID:Single] / URL: https://ditu.google.com
[G] Found Open Redirect. Payload: https://ditu.google.com?pleasedonthaveanamelikethis_plz_plz=pleasedonthaveanamelikethis_plz_plz
[G] Found Open Redirect. Payload: https://ditu.google.com
[G] Found Open Redirect. Payload: https://ditu.google.com?pleasedonthaveanamelikethis_plz_plz=pleasedonthaveanamelikethis_plz_plz
[G] Found Open Redirect. Payload: https://ditu.google.com
 ⠦  [SA: ✓ PA: ✓ ] Waiting for parameter and static analysis 

```

## Environment

 * Dalfox Version: v2.3.6
 * Installed from: go-get

**Comment by hahwul** [comment]
Thank you for submit issue!
I tested it in the same version and `--skip-bav` didn't deliver the actual bav request. but Perhaps the problem is caused by not checking `--skip-bav` at the 943~945 line I'll fix it quickly!

https://github.com/hahwul/dalfox/blob/1f32f3494e1aa3312f84b3e2a836eb61a9ae9aac/pkg/scanning/scan.go#L944

**Comment by hahwul** [comment]
- [ ] check `--skip-bav` option Or modified to check only in bav request
- [ ] Issue of detected unconditionally when scanning google domain

**Comment by hahwul** [comment]
Hello again @pdelteil 
I added code for #215 #217. I don't know if it was a good response, but I'd like to ask you to test it.

## Test
```
$ git clone https://github.com/hahwul/dalfox
$ cd dalfox
$ go build
```

version check
```
$ ./dalfox verison
v2.3.7-dev
```

Test...
```
$ ./dalfox ~~~
```

**Comment by pdelteil** [comment]
Hello there @hahwul. 

It works great!

**Comment by hahwul** [comment]
Closing the issue because released version 2.3.7. If there is any other problem, please re-open it! Enjoy your day!
